zipline/finance/trading.py

```diff
@@ -149,6 +153,49 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # stack.
         return False
 
+    def update_asset_finder(self,
```

Comment:
When would you need to update the asset finder in this manner rather than loading all the metadata in one shot?  If we can avoid mutating shared global state, we should do so.